packages/core/src/components/essentials/tooltip-bar.ts

```diff
@@ -165,23 +177,41 @@ export class TooltipBar extends Tooltip {
 			return { placement: TooltipPosition.BOTTOM, position: tooltipPos };
 		} else {
 			// positive bars
-			const tooltipPos = {
-				left:
-					barPosition.left -
-					holderPosition.left +
-					barPosition.width / 2,
-				top: barPosition.top - holderPosition.top - verticalOffset,
-			};
-
-			return { placement: TooltipPosition.TOP, position: tooltipPos };
+			if ((type === TooltipTypes.AXISLABEL && data.length > truncationThresholdProvided) ||
+				(type === TooltipTypes.LEGEND && data.name.length > truncationThresholdProvided)) {
+				const tooltipPos = {
+					left:
+						barPosition.left -
+						holderPosition.left +
+						barPosition.width * 2,
+					top: barPosition.top - holderPosition.top - verticalOffset,
+				};
+				return { placement: TooltipPosition.TOP, position: tooltipPos };
+			} else {
+				const tooltipPos = {
+					left:
+						barPosition.left -
+						holderPosition.left +
+						barPosition.width / 2,
+					top: barPosition.top - holderPosition.top - verticalOffset,
+				};
+				return { placement: TooltipPosition.TOP, position: tooltipPos };
+			}
 		}
 	}
 
 	/**
 	 * Returns the html for the bar single point tooltip
 	 * @param data associated values for the hovered bar
 	 */
-	getTooltipHTML(data: any) {
+	getTooltipHTML(data: any, type: TooltipTypes) {
+		if (type === TooltipTypes.LEGEND) {
+			const dataVal = data.name;
+			return `<div class="datapoint-tooltip"><p class="label">${dataVal}</p></div>`;
+		} else if (type === TooltipTypes.AXISLABEL) {
+			return `<div class="datapoint-tooltip"><p class="label">${data}</p></div>`;
+		}
```

Comment:
same as above, we either need to call the classes correctly by the tooltip type or have a trunction class for them all